File: backend/dabashou-order/src/main/java/com/dabashou/order/domain/Order.java
package com.dabashou.order.domain;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.dabashou.common.core.BaseEntity;

import java.time.LocalDateTime;

/**
 * 订单实体 — 对应 dbs_order 表(V1.0.0:127)
 * 继承 BaseEntity，获得 id / createTime / updateTime / deleted（逻辑删除）
 */
@TableName("dbs_order")
public class Order extends BaseEntity {

    private String orderNo;
    private Long buyerId;
    private Long sellerId;
    private Long demandId;
    private Long skillShelfId;
    private Long skillTagId;
    private String title;
    private Integer pointAmount;
    private Integer status;
    private String verifyCode;
    private LocalDateTime verifyCodeExpire;
    private Long timeSlotId;
    private LocalDateTime serviceStartTime;
    private LocalDateTime serviceEndTime;
    private LocalDateTime completeTime;
    private LocalDateTime cancelTime;
    private String cancelReason;
    private String remark;
    // id / createTime / updateTime / deleted 继承自 BaseEntity

    public String getOrderNo() { return orderNo; }
    public void setOrderNo(String orderNo) { this.orderNo = orderNo; }
    public Long getBuyerId() { return buyerId; }
    public void setBuyerId(Long buyerId) { this.buyerId = buyerId; }
    public Long getSellerId() { return sellerId; }
    public void setSellerId(Long sellerId) { this.sellerId = sellerId; }
    public Long getDemandId() { return demandId; }
    public void setDemandId(Long demandId) { this.demandId = demandId; }
    public Long getSkillShelfId() { return skillShelfId; }
    public void setSkillShelfId(Long skillShelfId) { this.skillShelfId = skillShelfId; }
    public Long getSkillTagId() { return skillTagId; }
    public void setSkillTagId(Long skillTagId) { this.skillTagId = skillTagId; }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public Integer getPointAmount() { return pointAmount; }
    public void setPointAmount(Integer pointAmount) { this.pointAmount = pointAmount; }
    public Integer getStatus() { return status; }
    public void setStatus(Integer status) { this.status = status; }
    public String getVerifyCode() { return verifyCode; }
    public void setVerifyCode(String verifyCode) { this.verifyCode = verifyCode; }
    public LocalDateTime getVerifyCodeExpire() { return verifyCodeExpire; }
    public void setVerifyCodeExpire(LocalDateTime verifyCodeExpire) { this.verifyCodeExpire = verifyCodeExpire; }
    public Long getTimeSlotId() { return timeSlotId; }
    public void setTimeSlotId(Long timeSlotId) { this.timeSlotId = timeSlotId; }
    public LocalDateTime getServiceStartTime() { return serviceStartTime; }
    public void setServiceStartTime(LocalDateTime serviceStartTime) { this.serviceStartTime = serviceStartTime; }
    public LocalDateTime getServiceEndTime() { return serviceEndTime; }
    public void setServiceEndTime(LocalDateTime serviceEndTime) { this.serviceEndTime = serviceEndTime; }
    public LocalDateTime getCompleteTime() { return completeTime; }
    public void setCompleteTime(LocalDateTime completeTime) { this.completeTime = completeTime; }
    public LocalDateTime getCancelTime() { return cancelTime; }
    public void setCancelTime(LocalDateTime cancelTime) { this.cancelTime = cancelTime; }
    public String getCancelReason() { return cancelReason; }
    public void setCancelReason(String cancelReason) { this.cancelReason = cancelReason; }
    public String getRemark() { return remark; }
    public void setRemark(String remark) { this.remark = remark; }
}

File: frontend/src/types/order.ts
/** 订单状态 */
export type OrderStatus = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7

export const OrderStatusMap: Record<OrderStatus, string> = {
  0: '已取消',
  1: '待支付',
  2: '已支付',
  3: '服务中',
  4: '待确认',
  5: '已完成',
  6: '已退款',
  7: '争议中',
}

export const OrderStatusColor: Record<OrderStatus, string> = {
  0: '#909399',
  1: '#e6a23c',
  2: '#409eff',
  3: '#FF8F00',
  4: '#e6a23c',
  5: '#67c23a',
  6: '#f56c6c',
  7: '#f56c6c',
}

/** 订单 */
export interface Order {
  id: number
  orderNo: string
  buyerId: number
  sellerId: number
  demandId: number | null
  skillShelfId: number | null
  skillTagId: number
  title: string
  pointAmount: number
  status: OrderStatus
  verifyCode: string | null
  verifyCodeExpire: string | null
  timeSlotId: number | null
  serviceStartTime: string | null
  serviceEndTime: string | null
  completeTime: string | null
  cancelTime: string | null
  cancelReason: string | null
  createTime: string
  updateTime: string
  // 前端附加字段
  buyerName?: string
  buyerAvatar?: string
  sellerName?: string
  sellerAvatar?: string
  tagName?: string
}

/** 积分流水 */
export interface PointTransaction {
  id: number
  userId: number
  orderId: number | null
  type: PointTransactionType
  amount: number
  balanceAfter: number
  description: string
  createTime: string
}

export type PointTransactionType = 1 | 2 | 3 | 4 | 5 | 6

export const PointTransactionTypeMap: Record<PointTransactionType, string> = {
  1: '收入',
  2: '支出',
  3: '冻结',
  4: '解冻',
  5: '系统奖励',
  6: '系统扣除',
}

/** 核销确认 */
export interface VerifyOrderParams {
  orderId: number
  verifyCode: string
}
